theano/sandbox/cuda/basic_ops.py

```diff
@@ -3329,20 +3329,42 @@ def c_code_cache_version(self):
         return ()
 
 
-class GpuFlatten(gof.HideC, tensor.Flatten, GpuOp):
+class GpuFlatten(gof.HideC, tensor.Reshape, GpuOp):
     """
     Implement Flatten on the gpu.
 
+    Note: The interface GpuFlatten is deprecated, you should use gpu_flatten
     """
 
     def make_node(self, x):
         assert isinstance(x.type, CudaNdarrayType)
-        rval = tensor.Flatten.make_node(self, x)
+        rval = flatten(x)
         host_out_broadcastable = rval.outputs[0].type.broadcastable
         out_type = CudaNdarrayType(broadcastable=host_out_broadcastable)
         return Apply(self, [x], [out_type()])
 
 
+
+def gpu_flatten(x, outdim=1):
+    """
+    Implement flatten on the gpu.
```

Comment:
It would be better to also explain what flatten is so the user doesn't have to look around for flatten to understand gpu_flatten.